components.html

```diff
@@ -0,0 +1,1381 @@
+          </button>
+        </div>
+      </div>
+
+
+
+      <!--COMPONENT LABEL-->
+      <p class="component-label">Textboxes</p>
+      <!--COMPONENT LABEL-->
+
+
+
+      <!--Textboxes-->
+      <input type="text" name="" class="text-black" placeholder="Placeholder text...">
+
+      <br /><br />
+
+      <input type="text" name="" class="text-black" placeholder="Readonly Placeholder text..." readonly>
+
+      <br /><br />
+
+      <textarea rows="3" name="" class="text-black" placeholder="Placeholder text..."></textarea>
+
+
+
+      <!--COMPONENT LABEL-->
+      <p class="component-label">Dropdown</p>
+      <!--COMPONENT LABEL-->
+
+
+
+      <!--Dropdown-->
+      <select>
+        <option value="">store.test1.lewis.brains.us</option>
+        <option value="">store.test2.lewis.brains.us</option>
+        <option value="">store.test3.lewis.brains.us</option>
+      </select>
+
+
+
+      <!--COMPONENT LABEL-->
+      <p class="component-label">Tooltip Overlays</p>
+      <!--COMPONENT LABEL-->
+
+
+
+      <!--Speech bubble (arrow left)-->
+      <div class="flex-container">
+        <div class="speech-shadow">
+          <div class="speech-bubble-left">
+            <p>Text goes here</p>
+          </div>
+        </div>
+      </div>
+
+
+
+      <br />
+
+
+
+      <!--Speech bubble (arrow top)-->
+      <div class="flex-container">
+        <div class="speech-shadow">
+          <div class="speech-bubble-top">
+            <p>Text goes here</p>
+          </div>
+        </div>
+      </div>
+
+
+
+      <br />
+
+
+
+      <!--Speech bubble (button tooltip)-->
+      <div class="flex-container">
+        <div class="speech-shadow">
+          <div class="speech-bubble-left">
+            <p class="margin-bottom">Are you sure?</p>
+
+            <!--Icon button (Toolip)-->
+            <div class="flex-container">
+              <button class="btn btn-tooltip">
+                <div class="flex-container">
+                  <div>
+                    <svg class="icon-16 btn-tooltip-icon">
+                      <use xlink:href="img/icons.svg#check-white" />
+                    </svg>
+
+                  </div>
+                  Save
+              </button>
+
+              <!--Icon button (Toolip outline)-->
+              <button class="btn-tooltip-stroke">
+                Cancel
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+
+
+
+      <br />
+
+
+
+      <!--Saved tooltip-->
+      <div class="flex-container">
+        <div class="tooltip-saved flex-container">
+          <svg class="icon-24 tooltip-saved-icon">
+            <use xlink:href="img/icons.svg#check-circular" />
+          </svg>
+          <p>Saved</p>
+        </div>
+      </div>
+
+
+
+      <!--COMPONENT LABEL-->
+      <p class="component-label">Search and Replace</p>
+      <!--COMPONENT LABEL-->
+
+
+
+      <!--Search and Replace-->
+      <table>
```

Comment:
In my current markup I'm avoiding using the table element as it's not really a dataset, which is what <table> is typically used for these days.

Reply:
@bradt Any ideas for this? I replaced the <table> tag with a div and it breaks the layout. I tried a few different things but couldn't get it to work.

Reply:
Again, I don't think semantics matter much here, so it's really whatever HTML and CSS are the best fit for the job. @ptasker What's your suggestion?

Reply:
In the placeholder markup I'm using CSS Grid with React's styled components  with a wrapper div as below:

```javascript
export const Row  = styled.div`
   display: grid;
   margin: 0.4rem 0;
   grid-template-columns: 0.5fr 5fr 0.5fr 5fr 0.5fr;
   row-gap: 10px;
   column-gap: 10px;
      .header-2{
          grid-column: 2
      }
      
      .header-4{
          grid-column: 4
      }
`;
```

This guy wraps 5 divs, and it creates the same layout as a table. Much less markup.
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout

I can make the styles you have here work with the existing markup I have, so no worries updating this.

Reply:
@ptasker Yes please. Then I'll have a look at it so I know for next time.